low-level/files/opt/seagate/sspl/setup/manifest_support_bundle

```diff
@@ -144,15 +144,104 @@ class ManifestSupportBundle(object):
+        node_support_data["hw_specifics"] = self.build_node_hw_specifics_instances()
+        self.write_content_in_file(node_support_data, self.node_manifest,
+                                   'node support data')
+        print('Node Support data Json Created Successfully..! :%s' % self.node_manifest)
+
+    def capture_enclosure_data(self):
+        # Get Enclosure Support Data
+        encl_support_data = {}
+        encl_support_data.update(self.support_common_data)
+        encl_data_response = self.build_encl_hw_specifics_instances()
+        if encl_data_response and 'enclosures' in encl_data_response:
+            encl_support_data["enclosure_wwn"] = encl_data_response["enclosures"][0]["enclosure-wwn"]
+            encl_support_data["hw_specifics"] = encl_data_response
+            self.write_content_in_file(encl_support_data, self.encl_manifest,
+                                        'enclosure support data')
+            print('Enclosure Support data Json Created Successfully..! :%s' % self.encl_manifest)
+        else:
+            print("Management controller IPs are not configured properly on node or in ConfStore", end=" ")
+            print("or it is inaccessible for the current node.")
+            print("Ignored to collect enclosure manifest data.")
+
+    def build_node_hw_specifics_instances(self):
+        lshw_dict = {}
+        with tempfile.TemporaryFile() as tempf:
+            proc = subprocess.Popen(['lshw', '-json'], stdout=tempf)
+            proc.wait()
+            tempf.seek(0)
+            str_dict = tempf.read().decode("utf-8")
+            lshw_dict = json.loads(str_dict)
+        return lshw_dict
+
+    def build_encl_hw_specifics_instances(self):
+        fan_module_list = []
+        sideplane_list = []
+        chassis_midplane_list = []
+        response = self.fetch_ws_request_data("/show/configuration", 'configuration')
+        if not response or response.status_code != singleton_realstorencl.ws.HTTP_OK:
+            return
+        enclosure_data = json.loads(response.text)
+        if enclosure_data and 'enclosure-fru' in enclosure_data and 'enclosures' in enclosure_data:
+            for encl_fru in enclosure_data['enclosure-fru']:
+                if encl_fru['name'] == 'FAN MODULE':
+                    try:
+                        fan_number = encl_fru['fru-location'].split()[-1]
+                        for fan_module in enclosure_data['enclosures'][0]['fan-modules']:
+                            if fan_module['durable-id'].split('.')[-1] == fan_number:
+                                fan_module['part-number'] = encl_fru['part-number']
+                                fan_module['serial-number'] = encl_fru['serial-number']
+                                fan_module_list.append(fan_module)
+                    except Exception as e:
+                        print("Error while updating fan module {0} information : {1}".format(fan_number, e))
```

Comment:
Should we proceed to generate other data if error happens in fan module info collection?

Reply:
build_encl_hw_specifics_instances() func should collect enclosure info for all the modules, 
then at #L208, there is a check, where only valid info will be parsed on enclosure_data dict.